nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestUpdateRecord.java

```diff
@@ -468,6 +470,67 @@ public void testSetRootWithUnescapeJsonCall() throws InitializationException, IO
+
+        runner.setProperty(jsonReader, SchemaAccessUtils.SCHEMA_ACCESS_STRATEGY, SchemaAccessUtils.SCHEMA_TEXT_PROPERTY);
+        runner.setProperty(jsonReader, SchemaAccessUtils.SCHEMA_TEXT, inputSchemaText);
+        runner.enableControllerService(jsonReader);
+
+        final JsonRecordSetWriter jsonWriter = new JsonRecordSetWriter();
+        runner.addControllerService("writer", jsonWriter);
+        runner.setProperty(jsonWriter, SchemaAccessUtils.SCHEMA_ACCESS_STRATEGY, SchemaAccessUtils.SCHEMA_TEXT_PROPERTY);
+        runner.setProperty(jsonWriter, SchemaAccessUtils.SCHEMA_TEXT, outputSchemaText);
+        runner.setProperty(jsonWriter, "Pretty Print JSON", "true");
+        runner.setProperty(jsonWriter, "Schema Write Strategy", "full-schema-attribute");
+        runner.enableControllerService(jsonWriter);
+
+        runner.setProperty(UpdateRecord.REPLACEMENT_VALUE_STRATEGY, UpdateRecord.RECORD_PATH_VALUES);
+        final Path input = Paths.get("src/test/resources/TestUpdateRecord/input/organisation.json");
+
+        // recursive conversion
+        runner.enqueue(input);
+        runner.setProperty("/", "unescapeJson(/record_json, 'true', 'true')");
+
+        runner.run();
+        runner.assertAllFlowFilesTransferred(UpdateRecord.REL_SUCCESS, 1);
+        String expectedOutput = new String(Files.readAllBytes(Paths.get("src/test/resources/TestUpdateRecord/output/organisation.json")));
+        runner.getFlowFilesForRelationship(UpdateRecord.REL_SUCCESS).get(0).assertContentEquals(expectedOutput);
+        assertTrue(runner.getLogger().getErrorMessages().isEmpty());
+
+        // no conversion
+        runner.clearTransferState();
+        runner.enqueue(input);
+        runner.setProperty("/", "unescapeJson(/record_json)");
+
+        runner.run();
+        runner.assertAllFlowFilesTransferred(UpdateRecord.REL_SUCCESS, 1);
+        expectedOutput = "[ {\n" +
+                "  \"name\" : null,\n" +
+                "  \"departments\" : null,\n" +
+                "  \"address\" : null\n" +
+                "} ]";
+        runner.getFlowFilesForRelationship(UpdateRecord.REL_SUCCESS).get(0).assertContentEquals(expectedOutput);
+        assertTrue(runner.getLogger().getErrorMessages().isEmpty());
+
+        // non-recursive conversion
+        runner.clearTransferState();
+        runner.enqueue(input);
+        runner.setProperty("/", "unescapeJson(/record_json, 'true')");
+        runner.setProperty(UpdateRecord.REPLACEMENT_VALUE_STRATEGY, UpdateRecord.RECORD_PATH_VALUES);
+
+        runner.run();
+        runner.assertAllFlowFilesTransferred(UpdateRecord.REL_FAILURE, 1);
+        final LogMessage errorMessage = runner.getLogger().getErrorMessages().get(0);
+        assertTrue(errorMessage.getMsg().contains("java.lang.ClassCastException: " +
+                "class java.util.LinkedHashMap cannot be cast to class org.apache.nifi.serialization.record.Record"));
```

Comment:
Checking for the text `ClassCastException` seems fair, but the text here is overly prescriptive. It could change at any time, and that should not cause failures of unit tests.